starterAIs/C# Starter

```diff
@@ -0,0 +1,122 @@
+            int initialResources = int.Parse(inputs[1]); // the initial amount of eggs/crystals on this cell
+            int neigh0 = int.Parse(inputs[2]); // the index of the neighbouring cell for each direction
+            int neigh1 = int.Parse(inputs[3]);
+            int neigh2 = int.Parse(inputs[4]);
+            int neigh3 = int.Parse(inputs[5]);
+            int neigh4 = int.Parse(inputs[6]);
+            int neigh5 = int.Parse(inputs[7]);
+
+            Cell cell = new Cell();
+            cell.index = i;
+            cell.cell_type = type;
+            cell.ressources = initialResources;
+
+            cell.neighbors = new int[6] { neigh0, neigh1, neigh2, neigh3, neigh4, neigh5 };
+            
+            cell.my_ants = 0;
+            cell.opp_ants = 0;
+
+            cells.Add(cell);
+        }
+        int numberOfBases = int.Parse(Console.ReadLine());
+
+        List<int> my_bases = new List<int>();
+        List<int> opp_bases = new List<int>();
+
+        inputs = Console.ReadLine().Split(' ');
+        for (int i = 0; i < numberOfBases; i++)
+        {
+            int myBaseIndex = int.Parse(inputs[i]);
+            my_bases.Add(myBaseIndex);
+        }
+        inputs = Console.ReadLine().Split(' ');
+        for (int i = 0; i < numberOfBases; i++)
+        {
+            int oppBaseIndex = int.Parse(inputs[i]);
+            opp_bases.Add(oppBaseIndex);
+        }
+
+        // game loop
+        while (true)
+        {
+            for (int i = 0; i < numberOfCells; i++)
+            {
+                inputs = Console.ReadLine().Split(' ');
+                int resources = int.Parse(inputs[0]); // the current amount of eggs/crystals on this cell
+                int myAnts = int.Parse(inputs[1]); // the amount of your ants on this cell
+                int oppAnts = int.Parse(inputs[2]); // the amount of opponent ants on this cell
+
+                cells[i].ressources = resources;
+                cells[i].my_ants = myAnts;
+                cells[i].opp_ants = oppAnts;
+            }
+
+            // Write an action using Console.WriteLine()
+            // To debug: Console.Error.WriteLine("Debug messages...");
+            List<string> actions = new List<string>();
+
+
+            // ------- ! -----   Now you can generate a list of actions to do --------
+            // WAIT | LINE <sourceIdx> <targetIdx> <strength> | BEACON <cellIdx> <strength> | MESSAGE <text>
+
+
+            // actions.Add("LINE 5 21 1");   
+
+
+            // WAIT | LINE <sourceIdx> <targetIdx> <strength> | BEACON <cellIdx> <strength> | MESSAGE <text>
+            if (actions.Count == 0)
+            {
+                Console.WriteLine("WAIT");
+            }
+            else
+            {
+                string final_action = null;
+
+
+
+                for (int i = 0; i < actions.Count; i++)
+                {
+                    final_action += actions[i];
+                    final_action += ";";
```

Comment:
could we replace this with a String.Join?

Reply:
Yes you right : 

string final_action = String.Join(";", actions);
Console.WriteLine(final_action);